RELEASE_CHECKLIST.md

```diff
@@ -4,16 +4,21 @@
-
 ## Release Process
 
+1. Check if there are any security vulnerabilities fixed and security advisories published before a release. Security advisories should be linked on the release notes.
 1. Determine a [SemVer2](https://semver.org/)-valid version prefixed with the letter `v` for release. For example, `version="v1.0.0-alpha.1"`.
-1. Bump up the `Version` in [internal/version/version.go](internal/version/version.go#L5) and open a PR for the changes.
-1. Wait for the PR merge.
-1. Be on the main branch connected to the actual repository (not a fork) and `git pull`.  Ensure `git log -1` shows the latest commit on the main branch.
-1. Create a tag `git tag -am $version $version`
-1. `git tag` and ensure the name in the list added looks correct, then push the tag directly to the repository by `git push --follow-tags`.
+1. If there are code changes in notation-go or notation-core-go library, follow the two steps above and cut the release for the notation library. Then update the dependency versions in notation `go.mod` and `go.sum`
+  - [go.mod](go.mod), [go.sum](go.sum)
+  - [test/e2e/go.mod](test/e2e/go.mod), [test/e2e/go.sum](test/e2e/go.sum)
+  - [test/e2e/plugin/go.mod](test/e2e/plugin/go.mod) and [test/e2e/plugin/go.sum](test/e2e/plugin/go.sum)
+1. Run `go mod tidy` locally to check if the `go.sum` file is also updated with any potential changes. Open a PR submit the changes to the notation repository. Please make sure this PR is merged with all E2E test cases passed before starting the next step. See [PR 754](https://github.com/notaryproject/notation/pull/754) as an example.
+1. Create another PR to update the Notation CLI version with a single commit when PRs in above steps are merged. The commit message MUST follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) and could be `bump: tag and release $version`. Record the digest of that commit as `<commit_digest>`. This PR is also used for voting purpose of the new release. Add the link of change logs and repo-level maintainer list in the PR's description. The PR title could be `bump: tag and release $version`. Make sure to reach a majority of approvals from the repo-level maintainers before releasing it. This PR should be merged using [Create a merge commit](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github) method in GitHub. See [PR 748](https://github.com/notaryproject/notation/pull/748) as an example. 
+1. After the version and dependencies are updated by the the PRs above in the [Notation](https://github.com/notaryproject/notation.git) repository, execute `git clone https://github.com/notaryproject/notation.git` to clone the repository to your local file system.

```

Comment:
After the vote PR being merged is sufficient.

Reply:
Updated